File: frontend/src/api/history.ts
import client from './client'
import type { HistoryItem, HistoryListResponse } from '../types'

export interface CreateHistoryData {
  task: string
  python_code: string
  file_name?: string | null
  summary?: string | null
  steps?: string[] | null
  tips?: string | null
  memo?: string | null
  exec_stdout?: string | null
  exec_stderr?: string | null
}

export async function getHistory(query?: string): Promise<HistoryListResponse> {
  const response = await client.get<HistoryListResponse>('/history', {
    params: query != null ? { q: query } : {},
  })
  return response.data
}

export async function getHistoryItem(id: string): Promise<HistoryItem> {
  const response = await client.get<HistoryItem>(`/history/${id}`)
  return response.data
}

export async function deleteHistory(id: string): Promise<void> {
  await client.delete(`/history/${id}`)
}

export async function createHistory(data: CreateHistoryData): Promise<HistoryItem> {
  const response = await client.post<HistoryItem>('/history', data)
  return response.data
}

File: backend/schemas/history.py
"""Schemas for the /api/history endpoints."""

from __future__ import annotations

from pydantic import BaseModel


class HistoryItem(BaseModel):
    id: str
    created_at: str
    task: str
    file_name: str | None
    summary: str | None
    python_code: str
    steps: list[str] | None  # parsed from JSON stored in DB
    tips: str | None
    memo: str | None
    exec_stdout: str | None
    exec_stderr: str | None


class HistoryListResponse(BaseModel):
    items: list[HistoryItem]
    total: int


class HistoryCreateRequest(BaseModel):
    task: str
    file_name: str | None = None
    summary: str | None = None
    python_code: str
    steps: list[str] | None = None
    tips: str | None = None
    exec_stdout: str | None = None
    exec_stderr: str | None = None


class HistoryUpdateMemo(BaseModel):
    memo: str
